server/service/microsoft_mdm.go

```diff
@@ -1274,7 +1274,23 @@ func (svc *Service) isFleetdPresentOnDevice(ctx context.Context, deviceID string
 	// If user identity is a MS-MDM UPN it means that the device was enrolled through user-driven flow
 	// This means that fleetd might not be installed
 	if isValidUPN(enrolledDevice.MDMEnrollUserID) {
-		return false, nil
+		var isPresent bool
+		if enrolledDevice.HostUUID != "" {
+			host, err := svc.ds.HostLiteByIdentifier(ctx, enrolledDevice.HostUUID)
+			if err != nil && !fleet.IsNotFound(err) {
+				return false, ctxerr.Wrap(ctx, err, "get host lite by identifier")
+			}
+			if host != nil {
+				orbitInfo, err := svc.ds.GetHostOrbitInfo(ctx, host.ID)
+				if err != nil && !fleet.IsNotFound(err) {
+					return false, ctxerr.Wrap(ctx, err, "get host orbit info")
+				}
+				if orbitInfo != nil {
+					isPresent = orbitInfo.Version != ""
```

Comment:
Anyone see a concern with this way of detecting if fleetd is installed? The `host_orbit_info` table is deleted when the `hosts` entry is deleted, so it should be fine on re-enrollment and things like that, but those re-enrollment flows can be tricky so I want to double-check this approach. Otherwise I think it is the best check possible, as it is data verified by the host itself (a bit like the "Verified" status on profiles).

Reply:
this makes sense to me, are you concerned about load? I think due to lack of APNs we make the hosts go this path very frequently.

not a blocker at all btw since this is a read

Reply:
Not too concerned about load, those are two reads that should be very quick (though I'm now semi-concerned by the `HostLiteByIdentifier`, I'll double-check if it is able to use an index). If it does end up being an issue, we could add a single-query approach so that it gets the orbit info in one go with the host uuid, but we already have tons of db accesses for hosts and related data so I try to reuse them when possible.

Reply:
My concern was more towards if `host_orbit_info` could get de-synchronized somehow after re-enrollments, saying that fleetd is installed when it isn't or something like that, but tbh I feel like in this case it would be a bug that we'd need to fix, and it would still be fine to read that information from `host_orbit_info` here.

Reply:
So our `Host...ByIdentifier` queries are doing full table scans, I'll update this to use something else...

```
mysql> explain  SELECT       h.id,       h.team_id,       h.osquery_host_id,       h.node_key,       h.hostname,       h.uuid,       h.hardware_serial,
h.distributed_interval,       h.config_tls_refresh,       COALESCE(hst.seen_time, h.created_at) AS seen_time     FROM hosts h     LEFT JOIN host_seen_times hst ON (h.id = hst.host_id) WHERE '1788FD73-B2D9-498B-BC96-181FD8C1DF95' IN (h.hostname, h.osquery_host_id, h.node_key, h.uuid, h.hardware_serial) LIMIT 1;
+----+-------------+-------+------------+--------+---------------+---------+---------+------------+------+----------+-------------+
| id | select_type | table | partitions | type   | possible_keys | key     | key_len | ref        | rows | filtered | Extra       |
+----+-------------+-------+------------+--------+---------------+---------+---------+------------+------+----------+-------------+
|  1 | SIMPLE      | h     | NULL       | ALL    | NULL          | NULL    | NULL    | NULL       | 2802 |   100.00 | Using where |
|  1 | SIMPLE      | hst   | NULL       | eq_ref | PRIMARY       | PRIMARY | 4       | fleet.h.id |    1 |   100.00 | NULL        |
+----+-------------+-------+------------+--------+---------------+---------+---------+------------+------+----------+-------------+
2 rows in set, 1 warning (0.00 sec)
```

Reply:
Oh actually, that won't change a thing - the bigger (and larger than just this use-case) issue is that we don't have an index on `hosts.uuid` :( I see that you already created [this ticket](https://github.com/fleetdm/fleet/issues/17963) to address this so I won't create another one. Will leave this PR like this and hopefully we can address this missing index soon enough.

Reply:
that sounds good, same for the `host_orbit_info` check 👍